### PR TITLE
Social Previews: Add Tumblr link preview

### PR DESCRIPTION
**Pull request by monsieur-z**
Related to 1204250791794927-as-1204281808721020

## Proposed Changes

This PR implements the Tumblr post preview.

### Prerequisites
- Make sure you have a Tumblr account.
- Make sure you have a self-hosted site connected to Jetpack, with a subscription that includes Social (e.g., Jetpack Complete).
- From the _Social_ or _My Jetpack_ page of your site WPadmin, activate Social.
- Then, connect your site to Tumblr (check [these instructions](https://jetpack.com/support/jetpack-social/connecting-to-social-networks/) if you need help).
- Spin up Calypso, by running this branch locally or using a live link below.

### Testing

- Create a new post from your site WPadmin
- In the Jetpack sidebar, make sure the post will be published automatically to your Tumblr page
Screenshot 2023-04-11 at 4 43 17 PM

- Publish it
- Visit `/posts/:site` in Calypso
- You should see the newly created post under the _Published_ tab
- In the post menu, click _Share_, then press the _Preview_ button
Screenshot 2023-04-11 at 4 47 11 PM

- Select the Tumblr preview
- You'll see a preview of how the post looks when shared

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/blocks/sharing-preview-pane/index.jsx [hunk 1]
```diff
@@ -124,7 +124,13 @@ class SharingPreviewPane extends PureComponent {
 					/>
 				);
 			case 'tumblr':
-				return <TumblrSharePreview { ...previewProps } />;
+				return (
+					<TumblrSharePreview
+						{ ...previewProps }
+						articleContent={ post.content }
```
Comment (monsieur-z) [inline client/blocks/sharing-preview-pane/index.jsx:130]: `post.content` is not truncated, which is what we want here: truncation is handled in the `social-previews` package.

File: client/components/share/tumblr-share-preview/index.jsx [hunk 2]
```diff
@@ -1,71 +1,36 @@
...
 			imageUrl,
-			translate,
+			message,
 		} = this.props;
 
-		const summary = truncateArticleContent( 396, articleContent );
+		const username = externalProfileURL?.match( /[^/]+$/ )[ 0 ];
+
 		return (
-			<div className="tumblr-share-preview">
```
Comment (monsieur-z) [inline client/components/share/tumblr-share-preview/index.jsx:23]: All this markup is not relevant anymore. Preview is implemented in the `social-previews` package.

**Review by andrii-lysenko — APPROVED** [review]
The code looks good to me and works as described! As I'm not yet have a full feature grasp, we could wait for other reviews as well.

Added a small comment regarding post images (not particularly for this PR) in the slack.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7965810

Thank you @monsieur-z for including a screenshot in the description! This is really helpful for our translators.

**Pull request merged**